### PR TITLE
Visual bugfixes and unexpected error

### PR DESCRIPTION
**Pull request by ikbensiep**
### Changes 
- Adds modified date to CollectionLinks (which might need to be renamed to CollectionListItems?)
- Adds client title to list item if available (ie, project list)
- Adds `.form-select` instead of `.form-control` in order to get a bootstrap `<select>` to show the little down arrow
- Adds NavLink instead of Link from `react-router. This allows main menu items to be `.active` on partial path matches (ie subpages) 
- Add some minor tweaks in `custom.css` (padding, hover, etc)

| enhanced list | dropdown | main menu highlight | 
| -------------------|----------------|-------------|
| image |  image | image |

### Question

I wanted to experiment with adding the new resource [client, person or project] form component on the list page (see Projects page). <br /> <br /> The action succeeds, however, this throws an error in in the useNewResource hook. <br /><br /> @evert  I'm curious if that's expected to happen in this case. 

 image

### CODE REVIEW

**Comment by evert** [comment]
Ah interesting! Exceptions should _never_ be thrown in hooks so this looks like a bug in useNewResource

File: src/components/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { Component, ErrorInfo } from 'react';
+
+interface State {
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<Record<string, any>, State> {
```
Comment (evert) [inline src/components/ErrorBoundary.tsx:8]: NICE

**Review by BeckyPollard — APPROVED** [review]
i enjoy

**Review by BeckyPollard — APPROVED** [review]
double approve

**Pull request merged**